src/routes/(app)/admin[[hash=admin_hash]]/product/[id]/+page.server.ts

```diff
@@ -205,16 +205,17 @@ export const actions: Actions = {
 					...(parsed.restrictPaymentMethods && {
 						paymentMethods: parsed.paymentMethods ?? []
 					}),
-					...(parsed.standalone && { hasVariations: parsed.hasVariations }),
+					...(parsed.standalone && {
+						hasVariations: Object.entries(cleanedVariationLabels?.names || []).length !== 0
+					}),
 					...(parsed.standalone &&
 						parsed.hasVariations && {
 							variations: parsed.variations?.filter(
 								(variation) => variation.name && variation.value
 							)
 						}),
 					...(parsed.standalone &&
-						parsed.hasVariations &&
-						parsed.variationLabels && {
+						parsed.hasVariations && {
```

Comment:
it's not consistent, you set `hasVariations` in DB based on other data but then set `variationLabels` based on `parsed.hasVariations`